### ISSUE TITLE
envoyfilter operation MERGE will generate double entry,if name is the same merge it ,not to generate two

### ISSUE THREAD

**Issue by 13567436138** [istio/istio#32413]
**Describe the feature request**
envoyfilter operation MERGE will generate double entry,if name is the same merge it ,not to generate two
for eg:
```
apiVersion: networking.istio.io/v1alpha3
kind: EnvoyFilter
metadata:
  name: access-log
  namespace: istio-system
spec:
  configPatches:
  - applyTo: NETWORK_FILTER
    match:
      listener:
        filterChain:
          filter:
            name: "envoy.filters.network.http_connection_manager"
    patch:
      operation: MERGE
      value:
        typed_config:
          "@type": "type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager"
          access_log:
          - name: envoy.access_loggers.file
            typed_config:
              "@type": "type.googleapis.com/envoy.extensions.access_loggers.file.v3.FileAccessLog"
              path: /dev/stdout
              log_format:
                text_format: "[%START_TIME%] \" %RESPONSE_CODE% \n"
```
the result is:

````
access_log: [
{
name: "envoy.access_loggers.file",
typed_config: {
@type: "type.googleapis.com/envoy.extensions.access_loggers.file.v3.FileAccessLog",
path: "/dev/stdout",
log_format: {
text_format: "[%START_TIME%] "%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL%" %RESPONSE_CODE% %RESPONSE_FLAGS% %RESPONSE_CODE_DETAILS% %CONNECTION_TERMINATION_DETAILS% "%UPSTREAM_TRANSPORT_FAILURE_REASON%" %BYTES_RECEIVED% %BYTES_SENT% %DURATION% %RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)% "%REQ(X-FORWARDED-FOR)%" "%REQ(USER-AGENT)%" "%REQ(X-REQUEST-ID)%" "%REQ(:AUTHORITY)%" "%UPSTREAM_HOST%" %UPSTREAM_CLUSTER% %UPSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_REMOTE_ADDRESS% %REQUESTED_SERVER_NAME% %ROUTE_NAME% "
}
}
},
{
name: "envoy.access_loggers.file",
typed_config: {
@type: "type.googleapis.com/envoy.extensions.access_loggers.file.v3.FileAccessLog",
path: "/dev/stdout",
log_format: {
text_format: "[%START_TIME%] " %RESPONSE_CODE% "
}
}
}
],
```
there are two log config with the same name.
**Describe alternatives you've considered**


[ ] Docs
[ ] Installation
[ x] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure



**Additional context**

**Comment by howardjohn** [comment]
This is working as expected, we do a protobuf MERGE operation, which for repeated fields means to append. It unfortunately doesn't work too well for this specific use case.

**Comment by ldemailly** [comment]
looks like this is fixed in 1.17, using `REPLACE` (https://istio.io/latest/news/releases/1.17.x/announcing-1.17/#added-support-for-filter-patching-in-istio) but any thoughts on how to configure something unique to some pods with 1.16?
